pkg/client/unversioned/restclient.go

```diff
@@ -72,15 +69,21 @@ func NewRESTClient(baseURL *url.URL, groupVersion unversioned.GroupVersion, c ru
 	base.RawQuery = ""
 	base.Fragment = ""
 
+	if config.GroupVersion == nil {
+		config.GroupVersion = &unversioned.GroupVersion{}
+	}
```

Comment:
As per the godoc of this type, group version should not be nil.  We
use empty group version to denote the absence of a version today.